R/download_data.R

```diff
@@ -0,0 +1,56 @@
+#' @title Download and Extract data from online sources
+#'
+#' @description Download and extract data from preset online sources, specifying
+#' the name of the dataset. 
+#' 
+#' @details This supersedes the method for pulling `hk_accidents` due to the
+#'   size of the data file.
+#'
+#' @param dataset String containing the name of the dataset. Valid values
+#'   include:
+#'     - `"hk_accidents"`
+#'     - `"hk_casualties"`
+#'     - `"hk_vehicles"`
+#'     
+#' @importFrom fst read_fst
+#' 
+#' @section Example:
+#' You can download and return a dataset with the following code:
+#' ```
+#' download_data(dataset = "hk_vehicles")
+#' ```
+#' 
+#' @export   
+download_data <- function(dataset = NULL){
+  
+  if(is.null(dataset)){
+    
+    stop("please provide the name of the dataset to pull.")
+    
+  }
```

Comment:
Would it be better to impose a stricter check of the dataset name? If the user typed the dataset name wrong, the `cannot open URL 'https://github.com/Hong-Kong-Districts-Info/hkdatasets/raw/master/data-ready/SOME-RANDOM-CHARS.fst'` error message may be hard to understand.

```suggestion
AVAILABLE_DATASETS = c("hk_accidents", "hk_casualties", "hk_vehicles")

if (!(dataset %in% AVAILABLE_DATASETS)) {
  stop(
    paste0("Please provide the name of the dataset to pull.\n",
           "Datasets currently available: ",
           # Convert column vector to text first
           paste0(AVAILABLE_DATASETS, collapse = ", ")
           )
  )
}
```

Reply:
That's a great suggestion. I'll add checks for the `dataset` argument input. Did this work for you locally? I like `fst` for its fast loading speed, so hopefully this will work as quite a scalable solution in the long term.

Reply:
Yep I ran the download and load fst data script in my local machine and it works well.

Reply:
Fantastic. I've now added some notes on `NEWS.md` and `README`, but this should be ready to merge to main and push to CRAN once ready.